library-best-practices/README.md

```diff
@@ -0,0 +1,19 @@
+Java Library Best Practices
+---------------------------
+
+The Java Library Best Practices are a collection of practices that we believe
+will minimize problems for consumers of interconnected Java libraries.  These
+practices come from decades of aggregated experience in maintaining open source
+Java libraries and are informed by many hard-earned lessons from mistakes that
+have been made. We have found that following these best practices will result in
+higher quality Java libraries with fewer diamond dependency conflicts and other
+kinds of problems. The list is open-ended, so new ones may be added from time to
+time.
+
+- [JLBP-1](JLBP-1.md): Minimize dependencies
+- [JLBP-2](JLBP-2.md): Minimize API surface
+- [JLBP-3](JLBP-3.md): Use Semantic versioning
+- [JLBP-4](JLBP-4.md): Avoid dependencies on unstable libraries and features
+- [JLBP-5](JLBP-5.md): Avoid dependencies that overlap classes with other
```

Comment:
classes or packages?

Reply:
"overlapping classes" is the way we describe the problem in JLBP-5 and JBLP-6; it's when multiple packages have the same fully-qualified class names.

Reply:
so it's not split packages, then? In that case, we probably do need a new JLBP to "not split packages" and we may want to reread 1-6 in this light. It's particularly important in Java 9 and later where split packages are now banned.

Reply:
Ok makes sense, I'll queue it up.